packages/client/src/StreamVideoClient.ts

```diff
@@ -47,14 +50,39 @@ export class StreamVideoClient {
 
   /**
    * You should create only one instance of `StreamVideoClient`.
-   * @param apiKey your Stream API key
-   * @param opts the options for the client.
    */
-  constructor(apiKey: string, opts?: StreamClientOptions) {
-    this.streamClient = new StreamClient(apiKey, {
-      persistUserOnConnectionFailure: true,
-      ...opts,
-    });
+  constructor(apiKey: string, opts?: StreamClientOptions);
+  constructor(args: {
+    apiKey: string;
+    options?: StreamClientOptions;
+    user?: User;
+    token?: TokenProvider;
+  });
+  constructor(
```

Comment:
I wanted to avoid breaking change, so the constructor accepts an object (new syntax) as well as api key + options (old syntax)